feat: add C implementation of `stats/base/dists/cauchy/logcdf` (#4390)
stdlib-js/stdlib: lib/node_modules/@stdlib/stats/base/dists/cauchy/logcdf/README.md

<!-- /.examples -->

<!-- C interface documentation. -->

* * *

<section class="c">

## C APIs

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/stats/base/dists/cauchy/logcdf.h"
```

#### stdlib_base_dists_cauchy_logcdf( x, x0, gamma )

> Evaluates the logarithm of the cumulative distribution function (logCDF) for a Cauchy distribution.

Reviewer: ```suggestion
Evaluates the natural logarithm of the [cumulative distribution function][cdf] (CDF) for a [Cauchy][cauchy-distribution] distribution with parameters `x0` (location parameter) and `gamma > 0` (scale parameter).
```

```c
double out = stdlib_base_dists_cauchy_logcdf( 4.0, 0.0, 2.0 );
// returns ~-0.16
```

The function accepts the following arguments:

-   **x**: `[in] double` input value.
-   **x0**: `[in] double` location parameter.
-   **gamma**: `[in] double` scale parameter.

```c
double stdlib_base_dists_cauchy_logcdf( const double x, const double x0, const double gamma );
```

</section>

<!-- /.usage -->

<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

</section>

<!-- /.notes -->

<!-- C API usage examples. -->

<section class="examples">

### Examples

```c
#include "stdlib/stats/base/dists/cauchy/logcdf.h"
#include "stdlib/constants/float64/eps.h"
#include <stdlib.h>
#include <stdio.h>

static double random_uniform( const double min, const double max ) {
    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
    return min + ( v*(max-min) );
}

int main( void ) {
    double gamma;
    double x0;
    double x;
    double y;
    int i;

    for ( i = 0; i < 25; i++ ) {
        x = random_uniform( 0.0, 10.0 );
        x0 = random_uniform( 0.0, 10.0 );

>         gamma = random_uniform( 0.0, 10.0 ) + STDLIB_CONSTANT_FLOAT64_EPS;

Reviewer: ```suggestion
        gamma = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS, 10.0 );
```

        y = stdlib_base_dists_cauchy_logcdf( x, x0, gamma );

>         printf( "x: %lf, k: %lf, γ: %lf, ln(F(x;x0,γ)): %lf\n", x, x0, gamma, y );

Reviewer: ```suggestion
        printf( "x: %lf, x0: %lf, γ: %lf, ln(F(x;x0,γ)): %lf\n", x, x0, gamma, y );
```
